main/src/ca/uwaterloo/flix/language/phase/PredDeps.scala

```diff
@@ -62,10 +62,10 @@ object PredDeps {
         case Type.Cst(TypeConstructor.Unit, _) => (Nil, den)
         case _ => (List(t), den) // Unary
       }
-    case _: Type.Var =>
-      // This could occur when querying or projecting a non-existent predicate
+    case _ =>
```

Comment:
This could be a bad type, but not Type.Error?

Should we instead try to ensure that bad types are always Type.Error?

Reply:
What do you mean? I don't understand.

Reply:
This default case covers the event where type inference failed. When that happens, I would hope that the type is `Type.Error` rather than some random type.

Reply:
How are you going to ensure that? It will be difficult because no only do we not know where to replace a type there may be multiple solutions.

Reply:
In any case thats beyond the scope of this PR